tests/cephadm/test_custom_container.py

```diff
@@ -0,0 +1,289 @@
+        cmd=f"curl -u {PRIVATE_REG_USER}:{PRIVATE_REG_PASS} \
+                https://{ceph.hostname}:5000/v2/_catalog"
+    )
+
+
+def run(ceph_cluster, **kw):
+    """Verify re-deploying of monitoring stack with custom images
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    # Get installer node
+    node = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Get config file and node
+    config = kw.get("config")
+    ceph_nodes = kw.get("ceph_nodes")
+
+    # Install httpsd-tools and httpd
+    node.exec_command(
+        sudo=True,
+        cmd="dnf install -y httpd-tools httpd",
+        long_running=True,
+    )
+
+    # Create folders for registry
+    node.exec_command(sudo=True, cmd="mkdir -p /opt/registry/{auth,certs,data}")
+
+    # Create credentials for accessing the private registry
+    node.exec_command(
+        sudo=True,
+        cmd=f"htpasswd -bBc {OPT_AUTH}/htpasswd \
+             {PRIVATE_REG_USER} {PRIVATE_REG_PASS}",
+    )
+
+    # Create a self-signed certificate
+    node.exec_command(
+        cmd=f'(echo "."; echo ".";echo ".";echo ".";echo ".";\
+                echo "{node.hostname}";echo ".";)| sudo openssl req -newkey \
+                rsa:4096 -nodes -sha256 -keyout {OPT_CERTS}/domain.key \
+                -x509 -days 365 -out {OPT_CERTS}/domain.crt -addext \
+                "subjectAltName = DNS:{node.hostname}"'
+    )
+
+    # Create a symbolic link to domain.cert
+    node.exec_command(
+        sudo=True,
+        cmd=f"ln -s {OPT_CERTS}/domain.crt \
+            {OPT_CERTS}/domain.cert",
+    )
+
+    # Add certificate to trusted list on the private registry node
+    node.exec_command(
+        sudo=True,
+        cmd=f"cp {OPT_CERTS}/domain.crt \
+            /etc/pki/ca-trust/source/anchors/",
+    )
+    node.exec_command(sudo=True, cmd="update-ca-trust")
+    update_cert(node)
+
+    # Perform passwordless ssh between all nodes
+    ceph_cluster.setup_ssh_keys()
+    file_name = f"{OPT_CERTS}/domain.crt"
+
+    # Copy the certificate to other nodes
+    for ceph in ceph_nodes:
+        ceph.exec_command(
+            sudo=True, cmd="touch /etc/pki/ca-trust/source/anchors/domain.crt"
+        )
+        contents, _ = node.exec_command(sudo=True, cmd=f"cat {file_name}")
+        key_file = ceph.remote_file(
+            sudo=True,
+            file_name="/etc/pki/ca-trust/source/anchors/domain.crt",
+            file_mode="a",
+        )
+        key_file.write(contents)
+        key_file.flush()
+        ceph.exec_command(sudo=True, cmd="update-ca-trust")
+        update_cert(node)
+
+    # Login to registry.redhat.io
+    con = get_cephci_config()
+    username_ = con.get("cdn_credentials", {}).get("username")
+    password_ = con.get("cdn_credentials", {}).get("password")
+    for ceph in ceph_nodes:
+        ceph.exec_command(
+            sudo=True,
+            cmd=f"podman login -u {username_} -p {password_} registry.redhat.io",
```

Comment:
Move this to - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/utils.py